### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by ArCheeq**
Adding routing and authorization to the application

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,14 +4,17 @@
   "private": true,
   "dependencies": {
     "@reduxjs/toolkit": "^1.9.3",
+    "axios": "^1.3.5",
     "bootstrap": "^5.2.3",
+    "oidc-client": "^1.11.5",
```
Comment (Misha999777) [inline package.json:9]: You can remove "oidc-client" and use just "oidc-client-ts" and "react-oidc-context"

File: src/auth/settings.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const settings = {
+    authority: process.env.REACT_APP_AUTHORITY,
+    client_id: process.env.REACT_APP_CLIENT_ID,
+    redirect_uri: process.env.REACT_APP_REDIRECT_URL,
```
Comment (Misha999777) [inline src/auth/settings.js:4]: I recommend using "window.location.href" as redirect_uri to return the user to the same page where authentication was requested

File: src/auth/settings.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+    response_type: process.env.REACT_APP_RESPONSE_TYPE,
+    scope: process.env.REACT_APP_SCOPE
+}
```
Comment (Misha999777) [inline src/auth/settings.js:7]: Please remove authentication data from the URL after successful authentication with something like this
```
onSigninCallback: () => {
    window.history.replaceState(
        {},
        document.title,
        window.location.pathname
    )
}
```

**Review by Misha999777 — CHANGES_REQUESTED** [review]
LGTM, except for issues mentioned in the comments.
Please resolve before merging.

File: package.json (hunk 1 above)
Comment (ArCheeq) [reply package.json]: Forgot to remove a dependency, originally used this library but had strange problems with it

File: src/auth/settings.js (hunk 3 above)
Comment (ArCheeq) [reply src/auth/settings.js:14]: Okay, I'll know how to do it right.

File: src/router/privateRoutes/PrivateRoutes.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Outlet } from "react-router-dom";
+import { useAuth } from "react-oidc-context";
+
+function PrivateRoutes() {
+    const auth = useAuth();
+    
+    if (!auth.isAuthenticated) {
+        auth.signinRedirect()
```
[the remark this answers is not in the document]
Comment (ArCheeq) [reply src/router/privateRoutes/PrivateRoutes.js]: fixed

**Review by Misha999777 — APPROVED** [review]
LGTM, thank you for addressing the issues.

**Pull request merged**